norduser/utmp.go

```diff
@@ -0,0 +1,94 @@
+package norduser
+
+/*
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <utmp.h>
+
+const int ERROR_REALLOC = -1;
+const int ERROR_MALLOC_USERNAME = -2;
+
+void free_users_table(char*** users, int size) {
+  for (int index = 0; index < size; index++) {
+    if ((*users)[index] != NULL) {
+      free((*users)[index]);
+    }
+  }
+
+  if (*users != NULL) {
+    free(*users);
+  }
+}
+
+int get_utmp_user_processes(char*** users) {
+  int index = 0;
+  int size = 0;
+  setutent();
+
+  *users = NULL;
+
+  struct utmp* u;
+  while ((u = getutent()) != NULL) {
+    if (u->ut_type != USER_PROCESS) {
+      continue;
+    }
+
+    if (index == size) {
+      char** tmp;
+      tmp = realloc(*users, (size + 1) * sizeof(char*));
```

Comment:
User name strings can be stored directly in the table:

    tmp = realloc(*users, (size + 1) * UT_NAMESIZE);

That way we would be able to get rid of per-user mallocs, only reallocs on the user table will remain.

Reply:
Yeah, but then we would just create a long string. An array of strings is more standard/idiomatic and easier to use on go side.